sklearnex/neighbors/common.py

```diff
@@ -68,14 +68,15 @@ def _onedal_supported(self, device, method_name, *data):
             if is_classifier:
                 class_count = 1
             is_single_output = False
-            if len(data) > 1 or hasattr(self, '_onedal_estimator'):
-                # To check multioutput, might be overhead
-                if len(data) > 1:
-                    y = np.asarray(data[1])
-                    if is_classifier:
-                        class_count = len(np.unique(y))
-                if hasattr(self, '_onedal_estimator'):
-                    y = self._onedal_estimator._y
+            y = None
+            # To check multioutput, might be overhead
+            if len(data) > 1:
+                y = np.asarray(data[1])
+                if is_classifier:
+                    class_count = len(np.unique(y))
+            if hasattr(self, '_onedal_estimator'):
+                y = self._onedal_estimator._y
+            if y is not None and hasattr(y, 'ndim') and hasattr(y, 'shape'):
                 is_single_output = y.ndim == 1 or y.ndim == 2 and y.shape[1] == 1
```

Comment:
What about the native python lists as an input?

Reply:
y is ndarray in all cases. It's converted to ndarray on training